apple/internal/apple_framework_import.bzl

```diff
@@ -52,9 +68,23 @@ def filter_framework_imports_for_bundling(framework_imports):
 
     return filtered_imports
 
-def _apple_framework_import_impl(ctx):
-    """Implementation for the apple_framework_import rule."""
-    framework_imports = ctx.files.framework_imports
+def _all_framework_binaries(frameworks_groups):
+    return [
+        _get_framework_binary_file(framework_dir, framework_imports.to_list())
+        for framework_dir, framework_imports in frameworks_groups.items()
+    ]
+
+def _get_framework_binary_file(framework_dir, framework_imports):
+    framework_name = paths.split_extension(paths.basename(framework_dir))[0]
+    framework_short_path = paths.join(framework_dir, framework_name)
+    for framework_import in framework_imports:
+        if framework_import.short_path == framework_short_path:
+            return framework_import
+
+    fail("ERORR: There has to be a binary file in the imported framework.")
+
+def _framework_dirs(framework_imports):
```

Comment:
rename to `_grouped_framework_files` as it's actually returning the map, and please edit the docstring to a description of this method (docstrings should fit in a single line, up to 100 characters with the quote chars, no need to expand to much)

Reply:
done